daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -128,21 +129,70 @@ private[lf] object Speedy {
     if (actual > limit)
       throw SError.SErrorDamlException(IError.Limit(error(limit)))
 
-  final case class OnLedger(
-      validating: Boolean,
-      submissionTime: Time.Timestamp,
-      contractKeyUniqueness: ContractKeyUniquenessMode,
+  final class OnLedgerMachine(
+      override val sexpr: SExpr,
+      override val traceLog: TraceLog,
+      override val warningLog: WarningLog,
+      override var compiledPackages: CompiledPackages,
+      override val profile: Profile,
+      val validating: Boolean,
+      val submissionTime: Time.Timestamp,
+      val contractKeyUniqueness: ContractKeyUniquenessMode,
       /* The current partial transaction */
       private[speedy] var ptx: PartialTransaction,
       /* Committers of the action. */
-      committers: Set[Party],
+      val committers: Set[Party],
       /* Additional readers (besides committers) for visibility checks. */
-      readAs: Set[Party],
+      val readAs: Set[Party],
       /* Commit location, if a scenario commit is in progress. */
-      commitLocation: Option[Location],
-      limits: interpretation.Limits,
-      disclosureKeyTable: DisclosedContractKeyTable,
-  ) extends LedgerMode {
+      val commitLocation: Option[Location],
+      val limits: interpretation.Limits,
+      val disclosureKeyTable: DisclosedContractKeyTable,
+  )(implicit loggingContext: LoggingContext)
+      extends Machine {
+
+    /** unwindToHandler is called when an exception is thrown by the builtin SBThrow or
```

Comment:
IIRC, Scaladoc does allow a `/** @inheritdoc */` style document comment to be used here. Suggesting this possibility as you may want to ensure the inherited comment and source comment remain in sync?

Reply:
The comment make sens only for the onLedger Machine. I remove from the declaration in `Machine`.